### ISSUE TITLE
setup.py build fails to correctly detect missing openssl on Mac OS X

### ISSUE THREAD

**Issue by joachimmetz** [VirusTotal/yara-python#17]
Darwin X 15.5.0 Darwin Kernel Version 15.5.0: Tue Apr 19 18:36:36 PDT 2016; root:xnu-3248.50.21~8/RELEASE_X86_64 x86_64

```
setup.py build
```

```
yara/libyara/modules/hash.c:17:10: fatal error: 'openssl/md5.h' file not found
#include <openssl/md5.h>
         ^
1 error generated.
```

**Comment by kenorb** [comment]
The same here.

**Comment by boolaz** [comment]
+1
Same issue with either "python setup.py build" or "pip install yard-python" on Mac OS X El Capitan

**Comment by zenGator** [comment]
Building yara 3.5.0 on Darwin [hostname].local 15.6.0 Darwin Kernel Version 15.6.0: Mon Aug 29 20:21:34 PDT 2016; root:xnu-3248.60.11~1/RELEASE_X86_64 x86_64
(el Capitan, v10.11.6) with Version 8.0 (8A218a)

Failed being unable to find openssl/md5.h

But on (fairly) identical Kernel version 14.5.0 (Yosemite, v10.10.5), I had no problem.

So, I moved the `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.10.sdk/usr/include/openssl` directory from the Yosemite box to my yara build directory and ran through `make clean  && ./bootstrap.sh && ./configure  && make && make install && printf "\n\nWin.\n"`

This was successful.  Builds and runs.

**Comment by kolmai** [comment]
Having the same issue, using `export CFLAGS='-I/usr/local/Cellar/openssl/1.0.2g/include/'` didn't help.
Anyone have an idea?

**Comment by boolaz** [comment]
The workaround I posted on 13th of september works fine on Yosemite.
But the result you get from yara search is slightly different than the one you get with Linux. 
I'll try to post an exhaustive how-to tomorrow.

**Comment by hi-T0day** [comment]
Any solutions?

**Comment by hi-T0day** [comment]
So,happy! I have found the solutions! It works in Mac OS 10.12
I install the openssl via brew install.
This way did not copy the include file to "usr/local/include".
I use this command to copy:

cp /usr/local/Cellar/openssl@1.1/1.1.0c/include/openssl  usr/local

It works!
@kenorb @boolaz

**Comment by brifordwylie** [comment]
$ pip install yara-python fails with this error on Mac Sierra... so anyone actively looking at this?

**Comment by 8u1a** [comment]
I had this issue on Sierra  (10.12.5), and then found this thread:
https://github.com/pyca/cryptography/issues/2692

tl;dr:
export LDFLAGS="-L/usr/local/opt/openssl/lib"
export CPPFLAGS="-I/usr/local/opt/openssl/include"
export PKG_CONFIG_PATH="/usr/local/opt/openssl/lib/pkgconfig"

I was then able to move past the clang error at #include <openssl/md5.h> and install yara-python with pip.

**Comment by wxsBSD** [comment]
I have a pull request up to fix this and another thing too.
